src/api/cvc4cpp.h

```diff
@@ -2559,6 +2559,26 @@ class CVC4_PUBLIC Solver
   Term mkTuple(const std::vector<Sort>& sorts,
                const std::vector<Term>& terms) const;
 
+  /**
+   * returns a term that represents ((_ tupSel index) tuple)
+   * @param index represents the index of the selected element in the tuple
+   * @param tuple a node that has a tuple type
+   * @pre sort of tuple is (Tuple T1 ... Tn) and index < n
+   * @return the element at the given index in tuple
+   */
+  Term tupleSelect(uint32_t index, Term tuple) const;
+
+  /**
+   * returns a tuple that represents
+   * (mkTuple ((_ tupSel i_1) t) ... ((_ tupSel i_n) t))
```

Comment:
```suggestion
   * returns a tuple ((_ project i_1 ... i_n) t) which note is equivalent to
   * (mkTuple ((_ tupSel i_1) t) ... ((_ tupSel i_n) t))
```

Reply:
There is some redundancy here. The implementation of this function actually returns a `tuple` term, not a `project` term. 
Currently If the user wants a project term, he should write
```cpp
  Op op = d_solver.mkOp(PROJECT, indices);
  Term term = d_solver.mkTerm(op, tuple);
```
The term would be rerwritten later to be the same as term returned by `Solver::tupleProject` .
It is unfortunate that `Solver::tupleProject` is similar to `Solver::tupleSelect` which is needed because we don't have a `tupleSelect` operator

Reply:
Rewriting is an internal detail for solving.  This documentation is for the purpose of the user, when you use this method, it should create a tuple projection term.  Thus it should say what is created, not what it internally rewrites to.